src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -180,37 +185,98 @@ protected override void ExecuteCore()
                 }
             }
 
-            if (!string.IsNullOrEmpty(AppHostRuntimeIdentifier) && !string.IsNullOrEmpty(appHostPackPattern))
+            AppHost = GetAppHostItem(appHostPackPattern, appHostRuntimeIdentifiers, appHostPackVersion,
+                packagesToDownload, AppHostRuntimeIdentifier, "AppHost");
+
+            if (PackAsToolShimAppHostRuntimeIdentifiers != null)
+            {
+                List<ITaskItem> packAsToolShimAppHostsList = new List<ITaskItem>();
+                foreach (var packAsToolShimAppHostRuntimeIdentifier in PackAsToolShimAppHostRuntimeIdentifiers)
+                {
+                    var PackAsToolShimAppHosts = GetAppHostItem(
+                            appHostPackPattern,
+                            appHostRuntimeIdentifiers,
+                            appHostPackVersion,
+                            packagesToDownload,
+                            packAsToolShimAppHostRuntimeIdentifier.ItemSpec,
+                            "PackAsToolShimAppHost");
+
+                    if (PackAsToolShimAppHosts != null)
+                    {
+                        packAsToolShimAppHostsList.AddRange(PackAsToolShimAppHosts);
+                    }
+                }
+                PackAsToolShimAppHosts = packAsToolShimAppHostsList.ToArray();
+            }
+
+            if (packagesToDownload.Any())
+            {
+                PackagesToDownload = packagesToDownload.ToArray();
+            }
+
+            if (runtimeFrameworks.Any())
+            {
+                RuntimeFrameworks = runtimeFrameworks.ToArray();
+            }
+
+            if (targetingPacks.Any())
+            {
+                TargetingPacks = targetingPacks.ToArray();
+            }
+
+            if (runtimePacks.Any())
+            {
+                RuntimePacks = runtimePacks.ToArray();
+            }
+
+            if (unresolvedFrameworkReferences.Any())
+            {
+                UnresolvedFrameworkReferences = unresolvedFrameworkReferences.ToArray();
+            }
+        }
+
+        private ITaskItem[] GetAppHostItem(
```

Comment:
Can this return `ITaskItem` instead of an array?  It's a little weird that it does return a single item as an array, solely so that we can assign straight to `AppHost` above.

Reply:
Apphost is ITaskItem[] there is existing comments. But i will make is return only ITaskItem
```
//  There should only be one AppHost item, but we use an item here so we can attach metadata to it
        //  (ie the apphost pack name and version, and the relative path to the apphost inside of it so
        //  we can resolve the full path later)
```